web/source/text/defaultOutput.ts

```diff
@@ -4,75 +4,122 @@
 /// <reference path="keyEvent.ts" />
 
 namespace com.keyman.text {
+  export enum EmulationKeystrokes {
+    Space = ' ',
+    Enter = '\n',
+    Backspace = '\b'
+  }
+
   /**
    * Defines a collection of static library functions that define KeymanWeb's default (implied) keyboard rule behaviors.
    */
   export class DefaultOutput {
     private constructor() {
     }
 
+    static codeForEvent(Lkc: KeyEvent) {
+      return Codes.keyCodes[Lkc.kName] || Lkc.Lcode;;
+    }
+
+    /**
+     * Serves as a default keycode lookup table.  This may be referenced safely by mnemonic handling without fear of side-effects.
+     * Also used by Processor.defaultRuleBehavior to generate output after filtering for special cases.
+     */
+    public static forAny(Lkc: KeyEvent, shiftState: number) {
+      var char = '';
+
+      let code = DefaultOutput.codeForEvent(Lkc);
+
+      // A pretty simple table of lookups, corresponding VERY closely to the original defaultKeyOutput.
+      if(char = DefaultOutput.forSpecialEmulation(Lkc)) {
+        return char;
```

Comment:
It still bothers me that we can return a `\b` for backspace, since it actually is never passed through to the app as such. I guess I don't have a complete understanding of the code paths for backspace handling.

Reply:
Added a comment here that may help:  https://github.com/keymanapp/keyman/pull/2854#discussion_r396211613

Or, well, it should be the next one you see on the PR's main page.